third_party/move/move-binary-format/src/deserializer.rs

```diff
@@ -1061,23 +1061,99 @@ fn load_signature_tokens(cursor: &mut VersionedCursor) -> BinaryLoaderResult<Vec
 
 fn load_function_attributes(
     cursor: &mut VersionedCursor,
+    bytecode_version: u32,
 ) -> BinaryLoaderResult<Vec<FunctionAttribute>> {
     let count = read_uleb_internal(cursor, ATTRIBUTE_COUNT_MAX)?;
     let mut attributes = Vec::with_capacity(count);
     for _ in 0..count {
-        attributes.push(load_attribute(cursor)?);
+        attributes.push(load_attribute(cursor, bytecode_version)?);
```

Comment:
Should we check that function attributes are not repeated (it looks like some of the code in the verifier assumes this invariant, but I am not sure where it is enforced)? Not sure if this is the canonical place, but asking here to make sure it is tracked.